### PR TITLE
test(http-api): add auth (password) tests

### PR DESCRIPTION
**Pull request by sunneydev**
added password auth tests for http_api which doesn't seem to be working

### CODE REVIEW

File: tests/dragonfly/http_conf_test.py [hunk 1]
```diff
@@ -75,3 +75,32 @@ async def test_password_on_admin(df_factory):
         async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("default", "XXX")) as session:
             resp = await session.get(f"http://localhost:{server.admin_port}/")
             assert resp.status == 200
+            
+async def test_no_password_on_http_api(df_factory):
+    with df_factory.create(
+        port=1112,
+    ) as server:
+        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("default", "XXX")) as session:
+            resp = await session.post(f"http://localhost:{server.port}/api", json={["ping"]})
```
Comment (romange) [inline tests/dragonfly/http_conf_test.py:84]: you send an invalid json. the right expression is `json=["ping"]`

File: tests/dragonfly/http_conf_test.py [hunk 2]
```diff
@@ -75,3 +75,32 @@ async def test_password_on_admin(df_factory):
...
+            assert resp.status == 200
+        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("random")) as session:
+            resp = await session.post(f"http://localhost:{server.port}/api", json={["ping"]})
```
Comment (romange) [inline tests/dragonfly/http_conf_test.py:87]: same here in all the places below

**Review by romange — CHANGES_REQUESTED** [review]
Thanks!  gave you some comments

File: tests/dragonfly/http_conf_test.py (hunk 1 above)
Comment (sunneydev) [reply tests/dragonfly/http_conf_test.py:84]: mb :D seems like i forgot python, haven't touched it for years.

**Comment by romange** [comment]
Thank you, @sunneydev . I can confirm we have some bugs there.
I can not submit this PR since it uncovers bugs but does not fix them. I opened #2863 where I copied your code and I will attribute your work as well.

**Comment by sunneydev** [comment]
> Thank you, @sunneydev . I can confirm we have some bugs there. I can not submit this PR since it uncovers bugs but does not fix them. I opened #2863 where I copied your code and I will attribute your work as well.

sure, closing this one than. thanks for addressing the bug.

**Pull request closed without merging**